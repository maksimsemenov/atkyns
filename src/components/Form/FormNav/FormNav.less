@import '~constants/styles.less';

.formNav {
  height: 100%;
  width: 400px;

  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  //border-right: 1px solid #eee;
}

.formNav__closeBtn {
  position: absolute;
  top: 20px;
  left: 20px;
  width: 40px;
  height: 40px;

  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    border-radius: 50%;
    transform: scale(0);
    opacity: 0;

    background-color: @color-light-blue;
    transition: transform @transition-basic, opacity @transition-basic;
  }
  &:hover::before {
    transform: scale(1);
    opacity: 1;
    //transition: transform @transition-active, opacity @transition-active;
  }
  &:active {
    transform: translateY(1px);
  }
}
.formNav__closeIcon {
  position: absolute;
  top: 10%;
  left: 10%;
  width: 80%;
  height: 80%;
  fill: @color-accent;
}
.formNav__list {
  margin: 0 0 40px 0;
  padding: 0;
  width: 100px;
  list-style-type: none;
  display: flex;
  flex-direction: column;
}
