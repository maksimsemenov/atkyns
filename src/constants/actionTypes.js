export const CHANGE_FIELD = 'CHANGE_FIELD'
export const ADD_CASE = 'ADD_CASE'

export const SET_STAGE = 'SET_STAGE'

export const SET_PAYMENT = 'SET_PAYMENT'
