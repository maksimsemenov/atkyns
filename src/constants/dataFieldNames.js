export const RELASHIONSHIP = 'relashionship'
export const REL_BY_ADOPTION = 'relByAdoption'
export const RES_THROUGH_ADOPTION = 'resThroughAdoption'

export const P_FIRST_NAME = 'petitioner/name/first'
export const P_MIDDLE_NAME = 'petitioner/name/middle'
export const P_FAMILY_NAME = 'petitioner/name/family'
export const P_OTHER_NAME = 'petitioner/name/other'
export const P_ADDRESS_STREET = 'petitioner/address/street'
export const P_ADDRESS_APT_NUMBER = 'petitioner/address/apartment'
export const P_ADDRESS_CITY = 'petitioner/address/city'
export const P_ADDRESS_STATE = 'petitioner/address/state'
export const P_ADDRESS_COUNTRY = 'petitioner/address/country'
export const P_ADDRESS_ZIP = 'petitioner/address/zip'
export const P_BIRTH_CITY = 'petitioner/BirthCity'
export const P_BIRTH_STATE = 'petitioner/BirthState'
export const P_BIRTH_COUNTRY = 'petitioner/BirthCountry'
export const P_BIRTH_DATE = 'petitioner/BirthDate'
export const P_GENDER = 'petitioner/Gender'
export const P_MARITAL_STATUS = 'petitioner/maritalStatus'
export const P_PMARRIAGE_DATE = 'petitioner/PMarriageDate'
export const P_PMARRIAGE_CITY = 'petitioner/PMarriageCity'
export const P_PMARRIAGE_STATE = 'petitioner/PMarriageState'
export const P_PMARRIAGE_COUNTRY = 'petitioner/PMarriageCountry'
export const P_SSN  ='petitioner/SSN'
export const P_ARN = 'petitioner/ARN'
export const P_PRIOR_MARRIAGES = 'petitioner/PriorMarriages'
export const P_STATUS_TYPE = 'petitioner/status/type'

export const R_FIRST_NAME = 'relative/name/first'
export const R_MIDDLE_NAME = 'relative/name/middle'
export const R_FAMILY_NAME = 'relative/name/family'
export const R_OTHER_NAME = 'relative/name/other'
export const R_NATIVE_NAME = 'relative/name/native'
