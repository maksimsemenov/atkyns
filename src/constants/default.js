export const price = 20;

export const STATUS_TYPE_CITIZEN = 'usCitizen'
export const STATUS_TYPE_PERMANENT = 'usPermanentResident'

export const HOST='http://localhost:5000'
