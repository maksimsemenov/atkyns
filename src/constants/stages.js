import * as fieldNames from 'constants/dataFieldNames'

export const STAGE_OVERVIEW = 'overview'
export const STAGE_PETITIONER = 'petitioner'
export const STAGE_RELATIVE = 'relative'
export const STAGE_PETITION = 'petition'

export const stages = [
  { name: STAGE_OVERVIEW },
  { name: STAGE_PETITIONER },
  { name: STAGE_RELATIVE },
  { name: STAGE_PETITION }
]

// const stages = {
//   '0': [
//     fieldNames.RELASHIONSHIP,
//     fieldNames.REL_BY_ADOPTION,
//     fieldNames.RES_THROUGH_ADOPTION
//   ]
// }
//
// export default stages
