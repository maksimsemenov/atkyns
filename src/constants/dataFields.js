/* @flow */
/* This file describes all data fields that we have in a project as well as relationships between them. Relationsips describes as effects that one field will cause on another
*/

import type { DataFieldT } from '../types/DataFieldT'

import * as fieldTypes from 'constants/fieldTypes'
import * as relTypes from 'constants/relationshipTypes'
import { STATUS_TYPE_CITIZEN, STATUS_TYPE_PERMANENT } from 'constants/default'
import * as DFN from 'constants/dataFieldNames'

const dataFields = new Map({
  [DFN.P_STATUS_TYPE]: {
    name: DFN.P_STATUS_TYPE,
    type: fieldTypes.OPTION,
    options: [ STATUS_TYPE_CITIZEN, STATUS_TYPE_PERMANENT ]
  },
  [DFN.RELASHIONSHIP]: {
    name: DFN.RELASHIONSHIP,
    value: '',
    type: fieldTypes.OPTION,
    disable: false,
    options: [
      { value: relTypes.SPOUSE },
      { value: relTypes.PARENT },
      { value: relTypes.BROTHER },
      { value: relTypes.CHILD },
    ],
    effects: [
      {
        fieldName: DFN.REL_BY_ADOPTION,
        effect: ({ value }) => ({
          disable: value === relTypes.SPOUSE ? true : false
        })
      }
    ]
  },

})[
  {
    name: DFN.RELASHIONSHIP,
    value: '',
    type: fieldTypes.OPTION,
    disable: false,
    options: [
      { value: relTypes.SPOUSE },
      { value: relTypes.PARENT },
      { value: relTypes.BROTHER },
      { value: relTypes.CHILD },
    ],
    effects: [
      {
        fieldName: DFN.REL_BY_ADOPTION,
        effect: ({ value }) => ({
          disable: value === relTypes.SPOUSE ? true : false
        })
      }
    ]
  },
  {
    name: 'relByAdoption',
    value: '',
    type: fieldTypes.BOOLEAN,
    disable: false
  },
  {
    name: 'resThroughAdoption',
    value: '',
    type: fieldTypes.BOOLEAN,
    disable: false
  },
  {
    name: 'pfirstName',
    value: '',
    type: fieldTypes.TEXT,
    disable: false
  }
]

export const petitionerStatus:

export default dataFields
