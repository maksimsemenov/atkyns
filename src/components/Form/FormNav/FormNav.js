import React, { PropTypes } from 'react'
import Link from 'react-router/Link'
import FormNavItem from 'components/Form/FormNavItem/FormNavItem'
import { STAGE_OVERVIEW, STAGE_PETITIONER, STAGE_RELATIVE, STAGE_PETITION } from 'constants/stages'
import './FormNav.less'

const FormCloseBtn = () =>
  <Link className='formNav__closeBtn' to='/'>
    <svg viewBox='0 0 30 30' className='formNav__closeIcon'>
      <rect x='8' y='8' width='6' height='6' rx='1' ry='1' />
      <rect x='16' y='8' width='6' height='6' rx='1' ry='1' />
      <rect x='8' y='16' width='6' height='6' rx='1' ry='1' />
      <rect x='16' y='16' width='6' height='6' rx='1' ry='1' />
    </svg>
  </Link>


const FormNav = ({ stages, caseId, stage, caseName }) => {
  return (
    <div className='formNav'>
      <div className='formNav__header'>
      </div>
      <FormCloseBtn />
      <ul className='formNav__list'>
        {stages.map((s, i) =>
          <FormNavItem
            key={i}
            stage={s.name}
            caseId={caseId}
            progress={stage.progress}
            active={s.name === stage}
          />
        )}
      </ul>
    </div>
  )
}
FormNav.propTypes = {
  stages: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    progress: PropTypes.number
  })).isRequired,
  caseId: PropTypes.string.isRequired,
  stage: PropTypes.oneOf([STAGE_OVERVIEW, STAGE_PETITIONER, STAGE_RELATIVE, STAGE_PETITION]).isRequired,
  caseName: PropTypes.string
}

export default FormNav
