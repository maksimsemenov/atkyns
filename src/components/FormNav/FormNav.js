import React, { PropTypes } from 'react'
import Link from 'react-router/Link'
import FormNavItem from 'components/FormNavItem/FormNavItem'
import './FormNav.less'

const FormCloseBtn = () =>
  <Link className='formNav__closeBtn' to='/'>
    <svg viewBox='0 0 30 30' className='formNav__closeIcon'>
      <rect x='8' y='8' width='6' height='6' rx='1' ry='1' />
      <rect x='16' y='8' width='6' height='6' rx='1' ry='1' />
      <rect x='8' y='16' width='6' height='6' rx='1' ry='1' />
      <rect x='16' y='16' width='6' height='6' rx='1' ry='1' />
    </svg>
  </Link>


const FormNav = ({ stages, caseId, caseName }) => {
  return (
    <div className='formNav'>
      <div className='formNav__header'>
      </div>
      <FormCloseBtn />
      <ul className='formNav__list'>
        {stages.map((stage, i) =>
          <FormNavItem
            key={i}
            stage={stage.name}
            caseId={caseId}
            progress={stage.progress}
          />
        )}
      </ul>
    </div>
  )
}
FormNav.propTypes = {
  stages: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    progress: PropTypes.number
  })).isRequired,
  caseId: PropTypes.string.isRequired,
  caseName: PropTypes.string
}

export default FormNav
