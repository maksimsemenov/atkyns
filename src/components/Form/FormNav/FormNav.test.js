import React from 'react'
import renderer from 'react-test-renderer'
import BrowserRouter from 'react-router/BrowserRouter'
import FormNav from './FormNav'
import { STAGE_OVERVIEW, STAGE_RELATIVE } from 'constants/stages'

describe('FormNav component', () => {
  it('renders without craches', () => {
    const stages = [
      {
        name: STAGE_OVERVIEW,
        progress: 20
      },
      {
        name: STAGE_RELATIVE,
        progress: 0
      }
    ]
    const tree = renderer
      .create(
        <BrowserRouter>
          <FormNav stages={stages} stage={STAGE_OVERVIEW} caseId='dsadajoiho' />
        </BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
