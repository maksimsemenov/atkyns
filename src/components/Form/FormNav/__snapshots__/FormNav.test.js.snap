exports[`FormNav component renders without craches 1`] = `
<div
  className="formNav">
  <div
    className="formNav__header" />
  <a
    className="formNav__closeBtn"
    href="/"
    onClick={[Function anonymous]}
    style={Object {}}>
    <svg
      className="formNav__closeIcon"
      viewBox="0 0 30 30">
      <rect
        height="6"
        rx="1"
        ry="1"
        width="6"
        x="8"
        y="8" />
      <rect
        height="6"
        rx="1"
        ry="1"
        width="6"
        x="16"
        y="8" />
      <rect
        height="6"
        rx="1"
        ry="1"
        width="6"
        x="8"
        y="16" />
      <rect
        height="6"
        rx="1"
        ry="1"
        width="6"
        x="16"
        y="16" />
    </svg>
  </a>
  <ul
    className="formNav__list">
    <li
      className="formNavItem">
      <a
        className="formNavItem__link"
        href="/case/dsadajoiho/overview"
        onClick={[Function anonymous]}
        style={Object {}}>
        %stage-overview
      </a>
    </li>
    <li
      className="formNavItem">
      <a
        className="formNavItem__link"
        href="/case/dsadajoiho/relative"
        onClick={[Function anonymous]}
        style={Object {}}>
        %stage-relative
      </a>
    </li>
  </ul>
</div>
`;
